ecl/hql/hqlfold.cpp

```diff
@@ -2154,6 +2154,82 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
             }
             break;
         }
+    case no_regex_findset:
+        {
+            IValue * t0 = expr->queryChild(0)->queryValue();
+            IValue * t1 = expr->queryChild(1)->queryValue();
+            IHqlExpression * c2 = queryRealChild(expr, 2);
+            IValue * t2 = c2 ? c2->queryValue() : NULL;
+            if (t0 && (!c2 || t2))
+            {
+                if(isUnicodeType(t0->queryType()))
+                {
+                    HqlExprArray results;
+                    unsigned plen = t0->queryType()->getStringLen();
+                    UChar * pattern = (UChar *)malloc((plen+1)*2);
+                    t0->getUCharStringValue(plen+1, pattern); //plen+1 so get null-terminated
+                    ICompiledUStrRegExpr * compiled = rtlCreateCompiledUStrRegExpr(pattern, !expr->hasAttribute(noCaseAtom));
+                    if (t1 && compiled)
```

Comment:
This check for t1 is too late.  If the second expression isn't a constant, this will cause the whole expression to be folded into an empty list.